### PR TITLE
New Version of OneWireBinding with many new features

### PR DESCRIPTION
**Pull request by Dennis650**
new features:
- individual refresh cyclus for each item
- value modifier, to modify values, like adjust/calibrate temperature values
- invertible contact and switch items
- AlarmSwitch for number values - it turns on, when a value is is greater then given maxWarning or less then minWarning
- item binding config has changed (see JavaDoc in AbstractOneWireDevicePropertyBindingConfig and Subclasses)

### CODE REVIEW

**Comment by Dennis650** [comment]
Tomorrow i will document the new item-configuration here: https://github.com/Dennis650/openhab/wiki/One-Wire-Binding---new-Version-%281.7.0-Snapshot%29

**Comment by teichsta** [comment]
Thanks a lot @Dennis650 for this contribution! I'll hope to find the time for the Review soon! Best, Thomas E.-E.

**Comment by staehler** [comment]
This binding enhancement isn't merged to openhab, right? My update yesterday deleted the preliminary binding from dennis ...

**Comment by teichsta** [comment]
no it isn't … will hurry up with the Review!

File: bundles/binding/org.openhab.binding.onewire/.classpath [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	<classpathentry kind="src" output="target/classes" path="src/main/java"/>
 	<classpathentry excluding="**" kind="src" output="target/classes" path="src/main/resources"/>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
-	<classpathentry kind="lib" path="lib/org.owfs.jowfsclient_0.9.0.jar"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+	<classpathentry kind="lib" path="lib/org.owfs.jowfsclient-1.2.6.jar" sourcepath="/Users/dennis/.m2/repository/org/owfs/jowfsclient/1.2.6/jowfsclient-1.2.6-sources.jar"/>
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.onewire/.classpath]: please remove the sourcepath

File: bundles/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/OneWireBindingProvider.java [hunk 2]
```diff
@@ -8,38 +8,40 @@
...
- * between openHAB items and OneWire items (sensors).
+ * This interface is implemented by classes that can provide mapping information between openHAB items and OneWire
+ * devices and their properties.
  * 
- * Implementing classes should register themselves as a service in order to be 
- * taken into account.
+ * Implementing classes should register themselves as a service in order to be taken into account.
  * 
- * @author Thomas.Eichstaedt-Engelen
+ * @author Dennis Riegelbauer
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/OneWireBindingProvider.java]: please don't remove other authors … just add yourself

File: bundles/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/OneWireActivator.java [hunk 3]
```diff
@@ -22,20 +21,20 @@
  */
 public final class OneWireActivator implements BundleActivator {
 
-	private static Logger logger = LoggerFactory.getLogger(OneWireActivator.class); 
-	
+	private static Logger LOGGER = LoggerFactory.getLogger(OneWireActivator.class);
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/OneWireActivator.java]: please either declare the logger "final" or revert this change. According to http://www.oracle.com/technetwork/java/codeconventions-135099.html only constants should be written in capital letters

**Comment by Dennis650** [comment]
Hi Thomas, i fixed the parts, you mentioned in your review notes.

**Comment by teichsta** [comment]
> Hi Thomas, i fixed the parts, you mentioned in your review notes.

**Comment by Dennis650** [comment]
Hi, need help on this. 
I found, there is a merge conflict: CONFLICT (content): Merge conflict in distribution/openhabhome/configurations/openhab_default.cfg
But where can i see, what exactly the problem is and how can i fix this? Thank you, Dennis

**Comment by teichsta** [comment]
try merge origin/master into your branch with ```git pull https://github.com/openhab/openhab.git master```. This command should fail with the same merge conflict. Then try ```git status``` to identify the file which contains the conflict.

Hope this helps,

Thomas E.-E.

**Comment by akileos** [comment]
Hi,
Seeing this update : " Dennis650 closed this 10 days ago "

Does that mean this won't be merged ?

Regards,

**Comment by Dennis650** [comment]
Hi,

i am new to git and github and destroyed my lokal repository trying to merge things. 

I made a new pull request: https://github.com/openhab/openhab/pull/2369

Greetings
Dennis
